### ISSUE TITLE
[bug] Unit test regression when compiling against OpenSSL 3.0.x

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#3041]
# Bug Report

**Current behavior:**

When testing custom builds on Fedora 36, I found that self-compiling any 1.14 source using OpenSSL 3(.0.5) will cause a regression in wallet crypto unit tests, caused by the code that compares our decryption with OpenSSL decryption. Our implementation is providing correct results, OpenSSL 3 is not.

Every decryption test fails with a line like:

```
wallet/test/crypto_tests.cpp(150): error: in "wallet_crypto/decrypt": 64cdd9d536ac9e95f2fcc5a7d84a295a0f801fe1efeaf9853d76ebf9fd416b00 != 64cdd9d536ac9e95f2fcc5a7d84a295a00000000000000000000000000000000
```

The errors do not occur when explicitly selecting openssl 1.1 system libraries, e.g. `openssl1.1-devel.x86_64` on Fedora 36. Or the default OpenSSL on i.e. Fedora 35 (as that is also 1.1)

**Note: this does not affect production code, only the unit test.**

**Expected behavior:**

Unit tests do not fail based on installed system software, if this software is not used inside the unit.

**Steps to reproduce:**

Dockerfile:
```Dockerfile
FROM fedora:36

ARG REF=1.14.6-dev
ARG REPO=https://github.com/dogecoin/dogecoin.git
ARG MAKE_OPTS=-j4

WORKDIR /testbuild

# install requirements (added git)
RUN dnf -y install gcc-c++ libtool which make autoconf automake openssl-devel \
                 libevent-devel boost-devel libdb-cxx-devel git

# clone and checkout ref
RUN git clone $REPO && \
    cd dogecoin && \
    git checkout $REF

WORKDIR /testbuild/dogecoin

# build: only disable bench, to save cycles
RUN ./autogen.sh && \
    ./configure --disable-bench && \
    make $MAKE_OPTS    

# test
RUN make check $MAKE_OPTS VERBOSE=1
```

**Dogecoin Core version**

Any 1.14.x source

**Machine specs**
- OS: Any linux that uses OpenSSL 3 

**Extra information**

[`gist:fedora36.log`](https://gist.github.com/patricklodder/bbd740e52705d49359edbb8b5e703506#file-fedora36-log)
[`gist:fedora35.log`](https://gist.github.com/patricklodder/bbd740e52705d49359edbb8b5e703506#file-fedora35-log)

**Comment by patricklodder** [comment]
Possible solution: Bitcoin Core removed this unit test process in https://github.com/bitcoin/bitcoin/pull/11024 - because its purpose is to make sure that our encryption and decryption code is compatible with wallets that used OpenSSL for encryption (pre-1.14 for us).

Additionally, we may want to add a bunch of static fixtures that we know work with both implementations (when not broken) and test these instead.

**Comment by patricklodder** [comment]
Also confirmed on Ubuntu 22.04 with Openssl 3.0.2

**Comment by patricklodder** [comment]
@envisiondata did you want to work on this one? If not, let me know then I push this through because this issue is growing (self-compiled builds on 2/3 recent linux I tried include openssl 3, so the problem is significant because automated builds on these platforms will fail on `make check`)

**Comment by envisiondata** [comment]
No My Linux box is slow. 
I'm going to work on https://github.com/dogecoin/dogecoin/issues/2908

**Comment by patricklodder** [comment]
Fixed with #3079
